### PR TITLE
FIX: Add onebox support for x.com

### PR DESCRIPTION
**Pull request by jancernik**

### CODE REVIEW

File: lib/onebox/engine/twitter_status_onebox.rb [hunk 1]
```diff
@@ -26,7 +26,13 @@ def to_html
       def get_twitter_data
         response =
           begin
-            Onebox::Helpers.fetch_response(url, headers: http_params)
+            # We need to allow cross domain cookies to prevent an
+            # infinit redirect loop between twitter.com and x.com
```
Comment (pmusaraj) [inline lib/onebox/engine/twitter_status_onebox.rb:30]:
```suggestion
            # infinite redirect loop between twitter.com and x.com
```

File: lib/onebox/helpers.rb [hunk 2]
```diff
@@ -113,9 +114,13 @@ def self.fetch_response(
...
-            # HACK: If this breaks again in the future, use HTTP::CookieJar from gem 'http-cookie'
-            # See test: it "does not send cookies to the wrong domain"
-            redir_header = { "Cookie" => cookie.join("; ") }
+            if allow_cross_domain_cookies
+              headers["Cookie"] = cookie.join("; ")
+            else
+              # HACK: If this breaks again in the future, use HTTP::CookieJar from gem 'http-cookie'
+              # See test: it "does not send cookies to the wrong domain"
+              redir_header = { "Cookie" => cookie.join("; ") }
+            end
```
Comment (pmusaraj) [inline lib/onebox/helpers.rb:123]: I'm not sure I follow this change... is this necessary?

**Review by pmusaraj — COMMENTED** [review]
Looks good overall, just one small question. 

We should add at least one test case using an `x.com` tweet URL to confirm that the twitter onebox engine is picked.

File: lib/onebox/helpers.rb (hunk 2 above)
Comment (jancernik) [reply lib/onebox/helpers.rb:123]: Do you mean the existing 'hack'? I haven't looked at it much, the spec even says 'unimplemented'

https://github.com/discourse/discourse/blob/7b437c9401a52cdda06402ebd9fff0a08b67ab2a/spec/lib/onebox/helpers_spec.rb#L176-L194

If you are referring to my change to allow cross-site cookie passing, yes. It took me a while to figure out, but without passing the cookies, x.com redirects to twitter.com, and twitter.com redirects back to x.com, and so on and so forth until we hit the redirect limit

**Review by pmusaraj — APPROVED** [review]
Let's follow up with tests in the next PR, but this is good to merge, thanks!

**Pull request merged**